src/ApiCallable.php

```diff
@@ -189,10 +238,20 @@ public static function createApiCall($stub, $methodName, CallSettings $settings,
         }
 
         if (array_key_exists('pageStreamingDescriptor', $options)) {
+            if ($isGrpcStreaming) {
+                throw new ValidationException(
+                    'grpcStreamingDescriptor not compatible with pageStreamingDescriptor'
+                );
+            }
             $apiCall = self::setPageStreaming($apiCall, $options['pageStreamingDescriptor']);
         }
 
         if (array_key_exists('longRunningDescriptor', $options)) {
+            if ($isGrpcStreaming) {
+                throw new ValidationException(
+                    'grpcStreamingDescriptor not compatible with longRunningDescriptor'
+                );
+            }
```

Comment:
There is a lot of mixing of validation and core logic; maybe the validation could be consolidated at the top to make the core flow cleaner?